v2/jrl-cmakemodules-v2/templates/__init__.py.in

```diff
@@ -0,0 +1,37 @@
+import os
+import platform
+
+class DllDirectoryManager():
+    """Context manager for managing DLL directories on Windows.
+    On Windows with Python 3.8+, Python doesn't search DLL in PATH anymore
+    We must specify DLL search path manually with `os.add_dll_directory`
+    On Linux and MacOS, we can use DYLD_LIBRARY_PATH and LD_LIBRARY_PATH
+    environment variables, or simply the rpath search for shared libraries.
+    ref: https://github.com/python/cpython/issues/87339#issuecomment-1093902060
+         https://docs.python.org/3/library/os.html#os.add_dll_directory
+    """
+
+    def safe_add_dll_directory(self, dll_dir: str):
+        """Add a DLL directory to the search path, if path exists."""
+        # Check if the directory exists
+        if os.path.isdir(dll_dir):
+            self.dll_dirs.append(os.add_dll_directory(dll_dir))
+
+    def __init__(self):
+        self.dll_dirs = []
+
+    def __del__(self):
+        for d in self.dll_dirs:
+            d.close()
+
+if platform.system() == 'Windows':
+    mgr = DllDirectoryManager()
+    module_path = os.path.dirname(__file__)
+    mgr.safe_add_dll_directory(os.path.join(module_path, '..', '..', '..', 'bin'))
+    dll_dirs = @__DLL_DIRS__@
+    for dll_dir in dll_dirs:
+        mgr.safe_add_dll_directory(os.path.join(module_path, dll_dir))
```

Comment:
I think `DllDirectoryManager.__del__` will never be called here.

mgr will never be delete when we import `@__MODULE_NAME__@`.

You can check it with `toto.py`:

```python
class Toto:
    def __init__(self):
        print("toto.__init__")

    def __del__(self):
        print("toto.__del__")


t = Toto()
```

If you `import toto` Only enter will be called.

You need to use contextlib to handle it well.